File: backend/app/schemas/folder.py
from datetime import datetime
from typing import Optional, List
from pydantic import BaseModel

class FolderBase(BaseModel):
    name: str
    parent_id: Optional[str] = None
    category_id: Optional[str] = None

class FolderCreate(FolderBase):
    pass

class FolderUpdate(BaseModel):
    name: Optional[str] = None
    category_id: Optional[str] = None

class FolderResponse(FolderBase):
    id: str
    level: int
    note_count: int = 0
    created_at: datetime

    class Config:
        from_attributes = True

class FolderTreeResponse(FolderResponse):
    children: List['FolderTreeResponse'] = []

File: frontend/src/types/folder.ts
export interface Folder {
  id: string
  name: string
  parent_id: string | null
  level: number
  category_id?: string | null
  note_count: number
  created_at: string
}

export interface FolderTree extends Folder {
  children: FolderTree[]
}

export interface FolderCreate {
  name: string
  parent_id?: string
  category_id?: string
}

export interface FolderUpdate {
  name?: string
  category_id?: string
}
